decentraland/helpers/index.js

```diff
@@ -81,40 +82,44 @@ export const addressesAreEqual = (address1, address2) => {
   return address1.toUpperCase() === address2.toUpperCase();
 };
 
-export const approveManaSpending = async fromAccount => {
+export const approveManaSpending = fromAccount => {
   const contracts = getContracts();
   const { manaContract, marketplaceContract } = contracts;
   const toAddress = marketplaceContract.getAddress();
   manaContract.setWallet(fromAccount);
+
   const action = manaContract.approve(toAddress, HALF_MILLION);
-  await action.waitForNonceToUpdate();
+  return action;
 };
 
-export const fundAccountWithEthers = async accountAddress => {
+export const fundAccountWithEthers = accountAddress => {
   const contracts = getContracts();
   const { gnosisSafeContract } = contracts;
   gnosisSafeContract.setWallet(gnosisSafeOwner);
   gnosisSafeContract.setSigners([gnosisSafeOwner]);
 
-  const transferEthersAction = gnosisSafeContract.transferEther(
-    accountAddress,
-    SMALL_AMOUNT
-  );
-  await transferEthersAction.waitForNonceToUpdate();
+  const action = gnosisSafeContract.transferEther(accountAddress, SMALL_AMOUNT);
+  return action;
 };
 
-export const fundAccountWithMana = async accountAddress => {
+export const fundAccountWithMana = accountAddress => {
   const contracts = getContracts();
   const { manaContract, gnosisSafeContract } = contracts;
   gnosisSafeContract.setWallet(gnosisSafeOwner);
   gnosisSafeContract.setSigners([gnosisSafeOwner]);
 
-  const transferManaAction = gnosisSafeContract.transferERC20(
+  const action = gnosisSafeContract.transferERC20(
     manaContract.getAddress(),
     accountAddress,
     HALF_MILLION
   );
-  await transferManaAction.waitForNonceToUpdate();
+  return action;
+};
+
+export const getTransactionHashFromAction = async action => {
+  const tx = await action.getTransaction();
```

Comment:
Oh cool, did we already have this function?

Reply:
Or are we just testing it for now using the testHelper mock action?

Reply:
Yes, we already have it.

Reply:
Great! Resolving this.